feature-dapp-impl/src/main/java/io/novafoundation/nova/feature_dapp_impl/presentation/main/DAppHeaderAdapter.kt

```diff
@@ -39,17 +73,56 @@ class DAppHeaderAdapter(val imageLoader: ImageLoader, val handler: Handler) : Re
         this.walletModel = walletModel
         notifyItemChanged(0, true)
     }
+
+    fun setFavorites(favoritesDApps: List<DappModel>) {
+        this.favoritesDApps = favoritesDApps
+        notifyItemChanged(0, true)
+    }
+
+    fun setCategories(categories: List<DAppCategoryModel>) {
+        this.categories = categories
+        notifyItemChanged(0, true)
+    }
+
+    fun showCategoriesShimmering(show: Boolean) {
+        showCategoriesShimmering = show
+        notifyItemChanged(0, true)
+    }
```

Comment:
We could use diff payloads to not rebind the whole header for each of those smaller changes